### PR TITLE
Add nearsphere convenience methods

### PR DESCRIPTION
**Pull request by mtrezza**
Adding convenience methods for queries with `geoWithin.nearSphere`.

Depending on https://github.com/parse-community/parse-server/pull/4825

### CODE REVIEW

File: src/__tests__/ParseQuery-test.js [hunk 1]
```diff
@@ -616,6 +616,63 @@ describe('ParseQuery', () => {
     });
   });
 
+  fit('can generate near-geopoint queries without sorting', () => {
```
Comment (flovilmart) [inline src/__tests__/ParseQuery-test.js]: Oops, needs to replace by ‘it’

**Comment by mtrezza** [comment]
@flovilmart good idea, introduced `sorted` parameter.
Should I make the `sorted` parameter nullable and default to `true` so that existing code does not break?

File: src/__tests__/ParseQuery-test.js (hunk 1 above)
Comment (mtrezza) [reply src/__tests__/ParseQuery-test.js]: done

**Comment by flovilmart** [comment]
Ok, tests are failing for now as they are run against an unpublished version of parse-server

**Comment by flovilmart** [comment]
I’ll do a release of the server later so we can get going with this one.

**Comment by flovilmart** [comment]
@mtrezza I’m wondering, can we make those tests succeed without the new parse-server? This way, we’ll be able to ship in a single server release with the JS SDK?

**Comment by mtrezza** [comment]
@flovilmart Tests against the generated query syntax in `src/__tests__/ParseQuery-test.js` pass without server update. Tests against the query results in `integration/test/ParseGeoPointTest.js` would require the server update AFAIK.

I think the actual query result tests are an important QA step so I suggest updating the server first. The update brings the full functionality with `withJSON` anyway, this PR is just adding convenience methods.

**Comment by mtrezza** [comment]
@flovilmart Yes, we could remove the tests against the query results (for now) in `integration/test/ParseGeoPointTest.js` and only keep the tests against the generated query syntax in `src/__tests__/ParseQuery-test.js`.

The query results are already tested in https://github.com/parse-community/parse-server/pull/4825 so if the query syntax is correct and passes the tests, it should yield the same query results. The integration tests behave a bit like chicken-and-egg it seems.

However, I think there should also be a test against query results of the convenience methods, so I'd add them ASAP later on. What do you suggest?

**Comment by flovilmart** [comment]
We can do the integrations tests in parse server instead, for this one or mark them xit or pending / optional

**Comment by mtrezza** [comment]
Yes, I already added the tests in parse server with https://github.com/parse-community/parse-server/pull/4827.

**Comment by mtrezza** [comment]
Yes, I already added the tests in parse server with https://github.com/parse-community/parse-server/pull/4827.

**Comment by mtrezza** [comment]
@flovilmart done

**Comment by dplewis** [comment]
> I’m wondering, can we make those tests succeed without the new parse-server? This way, we’ll be able to ship in a single server release with the JS SDK?

@flovilmart Can we do something similar to what we did with the PHP SDK and have it always go against the latest?

**Comment by dplewis** [comment]
@flovilmart Have you seen my last comment https://github.com/parse-community/Parse-SDK-JS/pull/582#issuecomment-397150539 

What do you think about using the latest parse server for testing?

**Comment by flovilmart** [comment]
We can definitely do that :) easier to catch regessions!

**Comment by mtrezza** [comment]
What do I have to change in code?

**Comment by dplewis** [comment]
@mtrezza You can start by changing the version of parse server used for test to the latest.

https://github.com/parse-community/Parse-SDK-JS/blob/master/integration/package.json#L6

to `github:parse-community/parse-server#latest`

Then remove your `xit` from test

Also I'm confused by change to the existing test cases if `sorted` is optional why are you forcing true? I feel like I'll have to change all my calls to sorted = true.

**Comment by mtrezza** [comment]
Change to the existing test cases was to make it apparent that there is a sorted parameter and to explicitly test with sorted=true. But it would surely be enough to have one case with sorted=true. So I’ll revert the existing test cases and add one for sorted=true.

**Comment by dplewis** [comment]
> So I’ll revert the existing test cases and add one for sorted=true.

That’s what I was going for. It looked like it was required

**Comment by mtrezza** [comment]
@dplewis Can you please have a look at the failing integration test, I think it is because parse-server#latest failed to build?

**Comment by dplewis** [comment]
@flovilmart According to the docs 
https://github.com/parse-community/parse-server#want-to-ride-the-bleeding-edge

I have to do `npm install parseplatform/parse-server.git#latest`

It currently fails for me 

```
npm ERR! prepareGitDep > babel src/ -d lib/ --copy-files
npm ERR! prepareGitDep 
npm ERR! prepareGitDep 
npm ERR! prepareGitDep 2> npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.
npm ERR! prepareGitDep src/ doesn't exist
```

`src/` folder doesn't exist in the latest branch

using `npm install parseplatform/parse-server.git#master` works

should the docs be updated or am I missing something

**Comment by dplewis** [comment]
@flovilmart How does this look?

On a side note should we pin the node packages? Don't want a repeat of https://github.com/parse-community/parse-server/issues/2040

**Comment by flovilmart** [comment]
@dplewis see my comment. I’ll have a look tomorrow too

**Comment by flovilmart** [comment]
@dplewis that sounds good! Thanks to this new NPM feature, one can use any branch and test against it. npm will fetch the dev deps and do it's thing! Wow :)

**Comment by dplewis** [comment]
@flovilmart thats pretty awesome now that I think about it.

**Comment by flovilmart** [comment]
@mtrezza Thanks again for the PR!

**Comment by mtrezza** [comment]
@flovilmart Sure thing, thanks to you and dplewis for the support.

**Comment by flovilmart** [comment]
Any time!

**Pull request merged**